src/common/snippets/src/lowered/loop_manager.cpp

```diff
@@ -222,18 +216,31 @@ void LinearIR::LoopManager::get_loop_bounds(const LinearIR &linear_ir,
         prev_loop_ids = (*std::prev(loop_begin_pos))->get_loop_ids();
     }
 
-    if (loop_ops_inserted) {
-        const auto loop_begin = ov::as_type_ptr<op::LoopBegin>((*std::prev(loop_begin_pos))->get_node());
-        OPENVINO_ASSERT(loop_begin, "Failed explicit loop bounds getting: LoopBegin has not been found");
-        const auto loop_end = loop_begin->get_loop_end();
-        OPENVINO_ASSERT(loop_end->get_id() == loop_id, "Failed explicit loop bounds getting: Loop bounds with correct ID have not been found");
-        loop_begin_pos = std::prev(loop_begin_pos);
-        loop_end_pos = linear_ir.find_after(loop_begin_pos, linear_ir.get_expr_by_node(loop_end));
-    } else {
-        // At the moment all Loops must have exit points
-        const auto& exit_expr = exits.back().expr_port->get_expr();
-        loop_end_pos = std::next(linear_ir.find_after(loop_begin_pos, exit_expr));
+    // At the moment all Loops must have exit points
+    const auto& exit_expr = exits.back().expr_port->get_expr();
+    loop_end_pos = std::next(linear_ir.find_after(loop_begin_pos, exit_expr));
+}
+
+void LinearIR::LoopManager::get_explicit_loop_bounds(const LinearIR &linear_ir, size_t loop_id,
+                                                     LinearIR::constExprIt &loop_begin_pos, LinearIR::constExprIt &loop_end_pos) const {
+    std::stack<LinearIR::constExprIt> loop_begin_poses;
+    loop_begin_pos = loop_end_pos = linear_ir.cend();
+    for (auto it = linear_ir.cbegin(); it != linear_ir.cend(); ++it) {
+        if (const auto loop_begin = ov::as_type_ptr<op::LoopBegin>((*it)->get_node())) {
+            loop_begin_poses.push(it);
```

Comment:
Do we really need to save iterators to all `LoopBegin`s we've seen?
It seems to get `LoopEnd` from every LoopBegin, and if this `LoopEnd` have a desired `loop_id`, we will increase iterator until we meet it.
This way we won't need to store all LoopBegin, and also we won't have to perform redundant checks for LoopEnds of nested loops.

Reply:
Agree that it's simpler. Done, thanks!